app/src/main/java/mozilla/lockbox/view/FxALoginFragment.kt

```diff
@@ -53,10 +53,13 @@ class FxALoginFragment : BackableFragment(), FxALoginView {
 
     override fun loadURL(url: String) {
         webView.webViewClient = object : WebViewClient() {
-            override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
-                webViewObserver?.accept(url)
-
-                super.onPageStarted(view, url, favicon)
+            override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
+                val url = request?.url.toString() ?: null
+                if ((presenter as FxALoginPresenter).isRedirectUri(url)) {
```

Comment:
Architecture-wise, this is a little too much knowledge about the `Presenter` from the `View` perspective. Rather than doing this force-cast and calling a function on the presenter, I would prefer to see the `FxAView` interface amended to look something like this:
```kotlin
interface FxALoginView {
    var webViewRedirect: ((url: URL?) -> Boolean)?
    val skipFxAClicks: Observable<Unit>?
    fun loadURL(url: String)
}
```

In the `onViewReady` of the `Presenter`, you can assign the anonymous function to carry out the same actions as the `webViewObserver` did, but additionally return the `isRedirectURI` value. The `shouldOverrideURLLoading` function would then be able to simply consist of 
```kotlin
return webViewRedirect(request?.url)
```